controllers/resource/upserter.go

```diff
@@ -125,12 +126,22 @@ func (u *upserter) UpsertDoguNetworkPolicies(ctx context.Context, doguResource *
 		multiErr = errors.Join(multiErr, fmt.Errorf("failed to create or update deny all rule for dogu %s: %w", dogu.GetSimpleName(), err))
 	}
 
-	for _, dependency := range dogu.Dependencies {
+	var allDependencies = dogu.Dependencies
+	for _, dep := range dogu.OptionalDependencies {
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1011: should replace loop with `allDependencies = append(allDependencies, dogu.OptionalDependencies...)`
<!-- __reviewdog__:ChBkNzAyMGMxMzAyMGFmNWFhEgpjaGVja3N0eWxl -->